Day3/TemperatureConversion.md

```diff
@@ -44,6 +44,112 @@ Create a new file, temperatureConversion.py. Copy the function definition code a
 
 
 
+------
+
+
+
+# Function: Quadratic roots
+
+A  [quadratic equation](https://en.wikipedia.org/wiki/Quadratic_equation) can be represented as
+
+$ax^2 + bx + c = 0$
+
+Such an equation has two roots, given by:
+
+$x = \frac{-b \pm \sqrt{b^2 - 4ac}}{2a}$
+
+
+
+We will write a Python program to ask the user to input the values of coefficients a, b and c. Then we will calculate and print the two values of x (the roots). For the purpose of simplicity and understanding in class, we will use a very simple equation with non-imaginary roots. 
+
+
+
+Let us consider the mathematical expression :$x = \frac{-b \pm \sqrt{b^2 - 4ac}}{2a}$
+
+In Python, this mathematical expression can be written as two separate expressions
+
+```python
+x1 = (-b + (b ** 2 - 4*a*c) ** 0.5) / (2 * a)
+x2 = (-b - (b ** 2 - 4*a*c) ** 0.5) / (2 * a)
+```
+
+Note that the ```**``` symbols is the exponentiation or "Raise to the power of" operator. Consequently, the square root of y, mathematically expressed as $\sqrt{y}$, is written in Python as ```y ** 0.5```.
+
+Alternatively, we can also use Python's built in Math library's ```math.sqrt()``` function, as shown below
+
+```python
+import math
+
+x1 =(-b + math.sqrt(b ** 2 - 4*a*c)) / (2 * a) 
+x2 =(-b - math.sqrt(b ** 2 - 4*a*c)) / (2 * a) 
+```
+
+
+
+To avoid clutter, we can further breakdown and write
+
+```python
+import math
+
+y = math.sqrt (b ** 2 - 4 *a*c)
+x1 = (-b + y) / (2 * a)
+x2 = (-b - y) / (2 * a)
+```
+
+
+
+Finally, we can put this into a function and write a program (e.g. quadroots.py)
+
+```python
+import math
+
+# Function to compute quadratic roots
+def quad_roots(a, b, c):
+    y = math.sqrt(b ** 2 - 4 * a * c)
+    x1 = (-b + y) / (2 * a)
+    x2 = (-b - y) / (2 * a)
+
+    # Return a list containing the two roots
+    return [x1, x2]
+
+# Main part of the program
+print("Solve for the quadratic equation ax^2 + bx + c = 0")
+a = input("Enter the value of a : ")
+b = input("Enter the value of b : ")
+c = input("Enter the value of c : ")
+
+# Convert the user inputs from string to float
+a = float(a)
+b = float(b)
+c = float(c)
+
+roots = quad_roots(a, b, c)
+print("Roots are:", roots[0], roots[1])
+```
+
+
+
+Test your program with the following inputs : a = 1, b = -8, c = 15. You should get roots of 5.0 and 3.0 (similar to what is shown below)
+
+```
+C:\Users\305002615\AppData\Local\Continuum\Anaconda3\python.exe C:/SHOURYA/DEV/PYTHON/summerschool/quadroots.py
```

Comment:
Since we are running in pycharm, let's remove the command line from your output.